TDSClient/TDSClient/AuthenticationProvider/ADALHelper.cs

```diff
@@ -0,0 +1,102 @@
+//  ---------------------------------------------------------------------------
+//  <copyright file="ADALHelper.cs" company="Microsoft">
+//     Copyright (c) Microsoft Corporation.  All rights reserved.
+//  </copyright>
+//  ---------------------------------------------------------------------------
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using TDSClient.TDS.Utilities;
+
+namespace TDSClient.AuthenticationProvider
+{
+    public class ADALHelper
+    {
+        private static readonly string AdoClientId = "4d079b4c-cab7-4b7c-a115-8fd51b6f8239";
+
+        /// NOTE: not possible to use username password with ADAL in netstandard 2.0 (.net framework possible)
+        /// <summary>
+        /// Gets JWT access token using ADAL with username and password.
+        /// </summary>
+        /// <param name="authority"></param>
+        /// <param name="resource"></param>
+        /// <param name="clientId"></param>
+        /// <param name="userId"></param>
+        /// <param name="password"></param>
+        /// <returns></returns>
+        // public static async Task<string> GetSQLAccessTokenFromADALUsingUsernamePassword(
+        //     string authority,
+        //     string resource,
+        //     string clientId,
+        //     string userId,
+        //     string password)
+        // {
+        //     UserPasswordCredential userCredentials = new UserPasswordCredential(userId, password);
+        //     return await GetAccessToken(authority, resource, clientId, userCredentials).ConfigureAwait(false);
+        // }
+
+        /// <summary>
+        /// Gets JWT access token using ADAL with integrated authentication.
+        /// </summary>
+        /// <param name="authority"></param>
+        /// <param name="resource"></param>
+        /// <returns></returns>
+        public static async Task<string> GetSQLAccessTokenFromADALUsingIntegratedAuth(
+            string authority,
+            string resource)
+        {
+            UserCredential userCredentials = new UserCredential();
+            return await GetAccessToken(authority, resource, userCredentials).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Gets access token using provided credential.
+        /// </summary>
+        /// <param name="authority"></param>
+        /// <param name="resource"></param>
+        /// <param name="credential"></param>
+        /// <returns></returns>
+        private static async Task<string> GetAccessToken(
+            string authority,
+            string resource,
+            UserCredential credential = null)
+        {
+            try
+            {
+                LoggingUtilities.WriteLog("Acquiring access token...");
+                AuthenticationContext authContext = new AuthenticationContext(authority);
```

Comment:
Is there an option to add Logger to this authContext so it logs whole output of ADAL calls?

Reply:
I will add the logger.

Reply:
For MSAL it's added.

Reply:
Added for ADAL too.